src/marqo/_httprequests.py

```diff
@@ -37,7 +38,14 @@ def _construct_path(self, path: str, index_name="") -> str:
         """Augment the URL request path based if telemetry is required."""
         base_url = self.config.instance_mapping.get_index_base_url(index_name=index_name) if index_name \
             else self.config.instance_mapping.get_control_base_url()
+
+        # Add v2 prefix if the request is sent to controller index API
+        if base_url.lower().startswith(os.environ.get("MARQO_CLOUD_URL", "https://api.marqo.ai")) and \
+                path.startswith("indexes"):
+            path = f"v2/{path}"
```

Comment:
nit: not a huge fan of mutating this var

would prefer something like 
```
if base_url...
    # control plane endpoints are now versioned
    url = f"{base}/v2/{path}"
else:
    # data plane endpoints are not 
    url = f"{base}/{path}"
```

Reply:
@wanliAlex did you test this?

Reply:
Yes, shallow tests